### PR TITLE
Updated the ActivityTypes to match the documentation

### PR DESCRIPTION
**Pull request by GitHub1530**
Added the COMPETING type, removed the WATCHING type and changed the order to match the documentation.

### CODE REVIEW

**Comment by JnCrMx** [comment]
Hmmm... It seems like the SDK is not up-to-date with the documentation.

The latest ``discord_game_sdk.h`` still defines the ActivityTypes as
```c
enum EDiscordActivityType {
    DiscordActivityType_Playing,
    DiscordActivityType_Streaming,
    DiscordActivityType_Listening,
    DiscordActivityType_Watching,
};
``` 
This seems to be an issue on Discord's side and you might wish to report it in https://github.com/discord/gamesdk-and-dispatch or https://github.com/discord/discord-api-docs. I sadly don't have time for that now.

Also, the new enum values now skip ``3`` (possibly in order to not break old versions): 
```
Playing | 0
Streaming | 1
Listening | 2
Custom | 4
Competing | 5
```
(https://discord.com/developers/docs/game-sdk/activities#data-models-activitytype-enum)

In order to reflect this in Java we need to add one "spacer" element to the enum (at the place where ``WATCHING`` was) or we get wrong values (e.g. ``COMPETING`` instead of ``CUSTOM``).
We might name this spacer either ``UNKNOWN``/``REMOVED``/``DEPRECATED``/etc. or just keep ``WATCHING`` and add an ``@Deprecated`` annotation (maybe together with one sentence of documentation) to it.

To be honest I'm not really sure about this ``COMPETING`` because I've never seen it "in the wild" before. But adding it probably won't hurt.

If you add this spacer value to the enum, I'll gladly merge this. Thank you for contributing :D

File: src/main/java/de/jcm/discordgamesdk/activity/ActivityType.java [hunk 1]
```diff
@@ -16,11 +16,6 @@
 	PLAYING,
 	STREAMING,
 	LISTENING,
-	WATCHING,
```
Comment (JnCrMx) [inline src/main/java/de/jcm/discordgamesdk/activity/ActivityType.java:19]: Please add a spacer enum to make up for skipped number.
See my comment to the pull request for more details.

**Review by JnCrMx — CHANGES_REQUESTED** [review]

File: src/main/java/de/jcm/discordgamesdk/activity/ActivityType.java (hunk 1 above)
Comment (GitHub1530) [reply src/main/java/de/jcm/discordgamesdk/activity/ActivityType.java:19]: I added a new value to the enum as a spacer.

**Comment by GitHub1530** [comment]
I added the spacer value (I just called it `UNUSED`), and I'm going to report it to the discord repos.
I hope the changes are ok now.

**Review by JnCrMx — APPROVED** [review]

**Comment by JnCrMx** [comment]
Yes, it looks fine now and my test runs without problems.

I will include some other new features in the next few days (there now seems to be an ``instance`` property in ``DiscordActivity``) and then make a new release (unless you need those changes immediately, then I will do the release right now, no problem).

Thank you again for contributing :D

**Pull request merged**